### ISSUE TITLE
Response codes between 200 < 300 are treated as HttpExceptions when they are indicating a success

### ISSUE THREAD

**Issue by free5ty1e** [agoda-com/ninjato#17]
# Bug report

## Describe the bug

I must manually intercept every Response object that won't expect a `200` for success and manually check response code and manually parse into intended Response body object, bypassing much of the convenience of the Ninjato library for what is a successful response, such as 201 or 204

## To Reproduce

Have a server return a success code higher than 200.  Only 200 will result in a "success" from Ninjato.

See this `Response` class, which is indeed `open` so I can subclass it, but the `isSuccess` field is *NOT* open so I cannot override this field which is used internally by Ninjato:

```
open class Response {
    lateinit var request: Request

    var code: Int = 200
    val headers: MutableMap<String, MutableList<String>> = mutableMapOf()
    var body: Body? = null

    val isSuccess: Boolean
        get() = code == 200

    /**
     * Factory class for creating instances of request.
     * In case you need to extend [Response] and enrich it with some data of your own,
     * provide instance of factory to [HttpClient].
     */
    abstract class Factory {
        abstract fun create(): Response
    }
}
```

## Expected behavior

I expect that any status code 200 - 299 would be considered a success, not only 200.  I would fix this in the Ninjato codebase by either opening the `isSuccess` field, or by changing the above `isSuccess` getter line to the following:

```
    val isSuccess: Boolean
        get() = code in 200..299
```

**Comment by free5ty1e** [comment]
For now, one workaround I'm using is to modify the `Response` inside an interceptor, and I'm simply changing any `2xx` response code to `200` before Ninjato sees it so that it treats it as a success.  This has the side effect of requiring any code processing inside the interceptor if I need to differentiate between a `200` and a `201` for some reason, for example.  But for most cases this works nice:

```
if (response.code in 200..299) {
    response.code = 200
}
return response

```

**Comment by Unlimity** [comment]
Hey there and thanks for reporting! I will fix this overlooked issue as soon as I have time to do so, but given that jCenter is shutting down and we still haven't figured out new publishing space I'm afraid you'll have to wait some time until you'll be able to grab the fix in the next revision.

**Comment by Unlimity** [comment]
Fix has been merged (#18)
